torch/fx/symbolic_trace.py

```diff
@@ -151,6 +151,21 @@ def search_for_tensor(m : torch.nn.Module, qualname_atoms : List[str]) -> Option
 def _proxy_placeholder(name: str, delegate: DelegateBase) -> Proxy:
     return Proxy(delegate.graph.placeholder(name), delegate)
 
+class SymVal:
+    pass
+
+def unpack_collection_type(proxy : Proxy, prototype : Any, delegate : DelegateBase):
+    if isinstance(prototype, list):
+        return [unpack_collection_type(proxy[i], prototype[i], delegate) for i in range(len(prototype))]
+    if isinstance(prototype, dict):
```

Comment:
I wonder whether we should generate some form of assert that the keys are always the same (and for lists that the lengths is constant). Not sure how represent those asserts in IR though